allennlp_models/generation/models/copynet_seq2seq.py

```diff
@@ -527,7 +536,13 @@ def _forward_loss(
         # shape: (batch_size,)
         log_likelihood = (log_likelihoods * target_mask).sum(dim=-1)
         # The loss is the negative log-likelihood, averaged over the batch.
-        loss = -log_likelihood.sum() / batch_size
+        if weight is None:
+            loss = -log_likelihood.sum() / batch_size
+        else:
+            # shape: (batch_size,)
+            if len(weight.shape) > 1:
+                weight = weight.squeeze()
+            loss = -(weight * log_likelihood).sum() / batch_size
```

Comment:
Shouldn't this be `/ weight.sum()`?

Reply:
I've gone back and forth on this.

If you use `weight.sum()` to normalize, then the weighting is only relative to each batch, which is probably *not* what you want.

For example, let's say we normalize by `weight.sum()` and your weights range from 0.5 - 1.0. If you have a batch that contains only instances with weights of 0.5, then this will give you the same result as if they all had weights of 1.0.

Reply:
Hmm, fair enough. I would expect that setting all the weights to 1000 should be the same as setting them all to 0.001. But to get that behavior, and also the behavior you want, we would have to sum up all the weights in the dataset before processing a single batch, and then scale each batch accordingly. That's not practical. So let's leave it like this then.